src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_black_knight.cpp

```diff
@@ -309,36 +323,56 @@ class npc_risen_ghoul : public CreatureScript
     {
         npc_risen_ghoulAI(Creature* creature) : ScriptedAI(creature)
         {
+            instance = creature->GetInstanceScript();
             Initialize();
         }
 
         void Initialize()
         {
             uiAttackTimer = 3500;
+            uiLeapTimer   = 1000;
         }
 
+        InstanceScript* instance;
         uint32 uiAttackTimer;
+        uint32 uiLeapTimer;
 
         void Reset() override
         {
             Initialize();
         }
 
+        void SpellHitTarget(Unit* /*victim*/, const SpellInfo* spell) override
+        {
+            if (spell->Id == SPELL_EXPLODE_H)
+                instance->SetData(DATA_I_VE_HAD_WORSE, false);
+        }
+
         void UpdateAI(uint32 uiDiff) override
         {
             if (!UpdateVictim())
                 return;
 
-            if (uiAttackTimer <= uiDiff)
+            if (uiLeapTimer <= uiDiff)
             {
-                if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1, 100, true))
+                if (Unit* target = SelectTarget(SELECT_TARGET_FARTHEST, 0, 30, true))
                 {
-                    if (target && target->IsAlive())
-                        DoCast(target, (SPELL_LEAP));
+                    DoResetThreat();
+                    me->AddThreat(target, 5.0f);
+                    DoCast(target, DUNGEON_MODE(SPELL_LEAP, SPELL_LEAP_H));
                 }
-                uiAttackTimer = 3500;
-            } else uiAttackTimer -= uiDiff;
 
+                uiLeapTimer = urand(7000, 10000);
+            }
+            else uiLeapTimer -= uiDiff;
+            
+            if (uiAttackTimer <= uiDiff)
+            {
+                DoCastVictim(DUNGEON_MODE(SPELL_CLAW, SPELL_CLAW_H));
```

Comment:
Use spelldifficulty_dbc if not already handled
http://collab.kpsn.org/display/tc/spelldifficulty_dbc